lib/components/user/after-signin-switch.js

```diff
@@ -12,17 +13,17 @@ import withLoggedInUserSupport from './with-logged-in-user-support'
  * Once state.user.loggedInUser is available:
  * - For new users, route them to the account page (it will show account setup).
  * - For existing users, route them to the itinerary search that was in place before login.
- * TODO: Improve this screen.
  */
 class AfterSignInSwitch extends Component {
   componentDidUpdate () {
     const { loggedInUser, routeTo } = this.props
+
+    // It will take several calls to componentDidUpdate() before loggedInUser becomes available.
     if (loggedInUser) {
-      if (isNewUser(loggedInUser)) {
-        routeTo('/account')
-      } else {
-        routeTo('/')
-      }
+      const nextPath = isNewUser(loggedInUser) ? '/account' : '/'
+
+      // Reroute and prevent the current URL from showing in the browser history.
+      routeTo(nextPath, null, replace)
```

Comment:
It's not visible in this screen, but we've picked up the practice to import actions (e.g., routeTo) in components in bulk so that we don't accidentally reference them outside of `this.props`. E.g.,:

```
import * as uiActions from '../../actions/ui'
...
...
const mapDispatchToProps = {
  routeTo: uiActions.routeTo
}
```

Reply:
Updated